packages/block-editor/src/components/inspector-controls/block-support-tools-panel.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * WordPress dependencies
+ */
+import { __experimentalToolsPanel as ToolsPanel } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useDispatch, useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../../store';
+import { cleanEmptyObject } from '../../hooks/utils';
+
+const labels = {
+	border: {
+		label: __( 'Border options' ),
+		header: __( 'Border' ),
+	},
+	color: {
+		label: __( 'Color options' ),
+		header: __( 'Color' ),
+	},
+	dimensions: {
+		label: __( 'Dimensions options' ),
+		header: __( 'Dimensions' ),
+	},
+	typography: {
+		label: __( 'Typography options' ),
+		header: __( 'Typography' ),
+	},
+};
+
+export default function BlockSupportToolsPanel( { group, children } ) {
+	const { clientId, attributes } = useSelect( ( select ) => {
+		const { getBlockAttributes, getSelectedBlockClientId } = select(
+			blockEditorStore
+		);
+		const selectedBlockClientId = getSelectedBlockClientId();
+
+		return {
+			clientId: selectedBlockClientId,
+			attributes: getBlockAttributes( selectedBlockClientId ),
+		};
+	} );
+	const { updateBlockAttributes } = useDispatch( blockEditorStore );
+
+	const resetAll = ( resetFilters = [] ) => {
+		const { style } = attributes;
+		let newAttributes = { style };
+
+		resetFilters.forEach( ( resetFilter ) => {
+			newAttributes = {
+				...newAttributes,
+				...resetFilter( newAttributes ),
+			};
+		} );
+
+		// Enforce a cleaned style object.
+		newAttributes = {
+			...newAttributes,
+			style: cleanEmptyObject( newAttributes.style ),
+		};
+
+		updateBlockAttributes( clientId, newAttributes );
+	};
+
+	return (
+		<ToolsPanel
+			label={ labels[ group ].label }
```

Comment:
I was wondering if we needed optional chaining here. We probably don't because by the time we're inside this function, there's already been a check that the `group` exists. But, to guard against accidentally missing a group in the list of labels, we could potentially use constants for the `labels` keys. Not a blocker for this PR since it's all working well, just thinking about making it easy make changes in follow-ups 🙂

Reply:
> We probably don't because by the time we're inside this function, there's already been a check that the group exists

That was my thinking. I'm happy to tighten this up though.

Reply:
I've added the optional chaining as suggested and replied in relation to the constants in this https://github.com/WordPress/gutenberg/pull/34157#discussion_r692687902.